lesson20/index.js

```diff
@@ -0,0 +1,134 @@
+const parent = document.querySelector("body");
+
+const createElementWithClassName = (type, className) => {
+    const element = document.createElement(type);
+    element.className = className;
+    return element;
+};
+
+const addLoading = () => {
+    const img = document.createElement("img");
+    const imgWrapper = createElementWithClassName("div", "loading");
+
+    img.src = "loading-circle.gif";
+    imgWrapper.id = "js-loading";
+    parent.appendChild(imgWrapper).appendChild(img);
+};
+
+const removeLoading = () => {
+    document.getElementById("js-loading").remove();
+};
+
+const fetchData = async(endpoint) => {
+    const response = await fetch(endpoint);
+
+    if(!response.ok){
+        const errorMessage = `${response.status}:${response.statusText}`;
+        parent.appendChild(createErrorMessage(errorMessage));
+        console.error(errorMessage);
+        return;
+    }
+    return await response.json();
+};
+
+const fetchUserData = async() => {
+    return new Promise((resolve) => {
+        setTimeout(() => resolve(fetchData("https://mocki.io/v1/8853f6f1-f1f0-4db3-9788-668933e2f788")), 3000);
+        // setTimeout(() => resolve(fetchData("https://mocki.io/v1/f12ae2d1-310d-4120-8749-47773d65e236")), 3000);//空配列
+        // setTimeout(() => resolve(fetchData("https://httpstat.us/503")), 3000);//503 error
+    })
+};
+
+const getUserData = async() => {
+    addLoading();
+    try {
+        const json = await fetchUserData();
+        if (!json) return;
+
+        const data = json.data;
+
+        if (data.length === 0) {
+            parent.textContent = "まだデータがありません";
+            console.log("まだデータがありません");
+        }
```

Comment:
IMO
I think it is necessary to write 
`parent.appendChild (createErrorMessage ("まだデータがありません")); `
for the processing when it is an empty array.
```suggestion
        if (data.length === 0) {
            parent.appendChild(createErrorMessage("まだデータがありません"));
            console.log("まだデータがありません");
            return; 
        }
```
result

<img width="514" alt="スクリーンショット 2022-05-08 11 12 30" src="https://user-images.githubusercontent.com/99260576/167279057-3a73af8a-e06f-4c96-b4fa-10a333bbd464.png">




With the current code,
 I can't do removeLoading (), so I get the following error:

<img width="544" alt="スクリーンショット 2022-05-08 11 15 04" src="https://user-images.githubusercontent.com/99260576/167278945-ed532bf8-2b19-4eb6-8f26-8f4ea1be74f2.png">

Reply:
Thanks for pointing that out!
So true...! You're right!
I had forgotten to check for errors ><
Fixed.
[8b0f145](https://github.com/haru-programming/JavaScript-lessons/pull/25/commits/8b0f145f8a4915e5fe8ea0b1a8f0a1361d6cbf98)